### PR TITLE
Adding rezz.lic

### PR DESCRIPTION
**Pull request by skyleronken**
Adding rezz.lic; rubocop completed.

### CODE REVIEW

**Comment by rpherbig** [comment]
> rubocop completed
❤️

File: rezz.lic [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    @settings = get_settings
+    rezz_prep(args.person)
+  end
+
+  def rezz_prep(person)
+    # cast buffs if set
+    check_buffs
+
+    # perceive body & rejuv
+    rejuv(person) unless is_silver?(person)
```
Comment (rpherbig) [inline rezz.lic]: What do you think about moving the `is_silver?(person)` check into `rejuv` as a guard clause?

File: rezz.lic [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      end
+    end
+    pause 1
+    waitrt?
+    UserVars.discerns[data['abbrev']] = discern_data
+    data['mana'] = discern_data['mana']
+    data['cambrinth'] = discern_data['cambrinth']
+  end
+
+  def calculate_mana(min, more, discern_data, cyclic)
```
Comment (rpherbig) [inline rezz.lic]: Is this copied from `crossing-training.lic`? (which has another copy in `gate.lic`)

File: rezz.lic [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        remaining -= step_size
+      end
+    else
+      discern_data['cambrinth'] = nil
+    end
+  end
+
+  def cast_spell(spell, mana, target, snap = false)
+    prepare?(spell, mana, false)
+    snap ? pause(5) : waitcastrt?
```
Comment (rpherbig) [inline rezz.lic]: This is already supported in `common-arcana` with the `prep_time` attribute on a spell. What do you think about using that?

File: rezz.lic [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+  def cast_soulbonding(person)
+    cast_spell('sb', 1, person, true)
+  end
+
+  def check_if_buff_active(spell, time)
+    DRSpells.active_spells[spell] && DRSpells.active_spells[spell] > time
+  end
+
+  # copied from crossing-trianing
```
Comment (rpherbig) [inline rezz.lic]: "training"

File: rezz.lic [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+      end
+    end
+  end
+
+  def is_silver?(person)
+    /As you concentrate, a thin (.*) nimbus/ =~ bput("perc #{person}", 'As you concentrate, a thin (.*) nimbus')
+    color = Regexp.last_match(1)
+    echo "Memories are #{color}"
+    waitrt?
+    return false unless color.eql? 'silver'
```
Comment (rpherbig) [inline rezz.lic]: You could instead return `color.eql? 'silver'`

File: rezz.lic [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+    wait_for_script_to_complete('buff', ['rezz'])
+  end
+
+  def skip_harness?
+    attunement = DRSkill.getrank('Attunement')
+    if attunement < 541
+      return false
+    elsif attunement > 600
+      return true
+      elseif check_if_buff_active('Persistence of Mana', 6)
```
Comment (rpherbig) [inline rezz.lic]: Is the duration of PoM based on your Attunement skill? I noticed that if your skill is < 541 it won't check for PoM

File: rezz.lic (hunk 1 above)
Comment (skyleronken) [reply rezz.lic]: done!

File: rezz.lic (hunk 2 above)
Comment (skyleronken) [reply rezz.lic]: I have no problem with this. If I recall correctly there was an issue with doing this. I think it was a business rule. Something about yaml settings in common libs. I can move this to common-arcana if that is okay.

File: rezz.lic (hunk 3 above)
Comment (skyleronken) [reply rezz.lic]: the common-arcana cast_spell function does not support specifying a target to cast against.

File: rezz.lic (hunk 5 above)
Comment (skyleronken) [reply rezz.lic]: done!

File: rezz.lic (hunk 6 above)
Comment (skyleronken) [reply rezz.lic]: With rezz you dont need to harness mana before infusing a spell if the following rules apply:
- You have more than 600 attunement
OR
- You have less than 600 but more than 541 attunement AND have POM active.

File: rezz.lic (hunk 4 above)
Comment (skyleronken) [reply rezz.lic]: hah! fixed

File: rezz.lic (hunk 3 above)
Comment (rpherbig) [reply rezz.lic]: I think it will work if you `spell['cast] = "cast #{target}"` beforehand

Comment (skyleronken) [reply rezz.lic]: Okay, duh. I made the changes.

File: rezz.lic (hunk 2 above)
Comment (skyleronken) [reply rezz.lic]: i made the changes. going to test a little. Need to find a dead person though lol.

Comment (rpherbig) [reply rezz.lic]: Ah, you're right. It's the `prep_scaling_factor` and `cambrinth_cap` settings. I think we had much the same conversation in #2099 not too long ago.

Let's leave the copy/paste in for _this_ PR, but I would very much like to see the next one be extracting that logic. The catch would be that those two settings have to be passed in as parameters, not pulled out of settings like they are today.

File: rezz.lic (hunk 6 above)
Comment (rpherbig) [reply rezz.lic]: Got it, thank you

File: rezz.lic (hunk 2 above)
Comment (skyleronken) [reply rezz.lic]: Im still going to test adding to common-arcana. I followed the same methodology in other common-arcana functions that require settings but providing it as an argument to the function. So you can call check_discern(data, settings) now which in turn will pass to calculate_mana.

Comment (jandersson) [reply rezz.lic]: Took care of the commonizing of discern, the signature is as you mention
`data = check_discern(data, settings)`

**Comment by skyleronken** [comment]
@rpherbig latest commit includes all of the discussed changes

However, it is dependent upon @jandersson DRCA discerns PR which is pending its co-dependent cast_spell update. Once thats resolved this should be good to g.

**Pull request merged**